proteus/mprans/SW2DCV.h

```diff
@@ -2039,7 +2028,7 @@ namespace proteus
 	  // CALCULATE EDGE BASED CFL //
 	  //////////////////////////////
 	  double mi = lumped_mass_matrix[i];
-	  edge_based_cfl[i] = fabs(dLowii)/mi;
+	  edge_based_cfl[i] = 2*fabs(dLowii)/mi;
```

Comment:
Is this  a safety factor?

Reply:
We don't know the exact condition on the time step so that the "low-order" method is stable. The paper we consider for that method states that there is a constant C such that if dt <= C*dLii/mi for all i then the method preserves the monotonicity constraints. Then having or not the 2 factor is irrelevant. At the end of the day we have to recompute the time step if positivity is violated.